extensions/binding/org.eclipse.smarthome.binding.mqtt/src/main/java/org/eclipse/smarthome/binding/mqtt/internal/discovery/MqttServiceDiscoveryService.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.smarthome.io.transport.mqtt.MqttServiceObserver;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MqttServiceDiscoveryService} is responsible for discovering connections on
+ * the MqttService shared connection pool.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(immediate = true, service = DiscoveryService.class, configurationPid = "org.eclipse.smarthome.MqttServiceDiscoveryService")
+public class MqttServiceDiscoveryService extends AbstractDiscoveryService implements MqttServiceObserver {
+    private final Logger logger = LoggerFactory.getLogger(MqttServiceDiscoveryService.class);
+    MqttService mqttService;
+
+    public MqttServiceDiscoveryService() {
+        super(Stream
+                .of(MqttBrokerBindingConstants.BRIDGE_TYPE_SYSTEMBROKER, MqttBrokerBindingConstants.BRIDGE_TYPE_BROKER)
+                .collect(Collectors.toSet()), 500, true);
+    }
+
+    @Override
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        super.activate(config);
+    };
+
+    @Override
+    @Deactivate
+    protected void deactivate() {
+        super.deactivate();
+    }
+
+    @Reference
+    public void setMqttService(MqttService service) {
+        mqttService = service;
+    }
+
+    public void unsetMqttService(MqttService service) {
+        mqttService = null;
+    }
+
+    @Override
+    protected void startScan() {
+        mqttService.addBrokersListener(this);
+        mqttService.getAllBrokerConnections().forEach((brokerId, broker) -> brokerAdded(brokerId, broker));
+        stopScan();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (mqttService == null) {
+            return;
+        }
+        mqttService.addBrokersListener(this);
+        mqttService.getAllBrokerConnections().forEach((brokerId, broker) -> brokerAdded(brokerId, broker));
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (mqttService == null) {
+            return;
+        }
+        mqttService.removeBrokersListener(this);
+    }
+
+    @Override
+    public void brokerAdded(String brokerId, MqttBrokerConnection broker) {
+        logger.trace("Found broker connection {}", brokerId);
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("host", broker.getHost());
+        properties.put("port", broker.getPort());
+        ThingUID thingUID;
+        thingUID = new ThingUID(MqttBrokerBindingConstants.BRIDGE_TYPE_SYSTEMBROKER, brokerId);
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                .withLabel("Mqtt Broker (" + brokerId + " - " + broker.getHost() + ")").build());
```

Comment:
It seems a bit unfortunate that MqttBrokerConnections do not have a label to use for UIs - imho we will need such a property in order to properly deal with them in UIs, wdyt?

You should remove the parts in brackets from the label - we instead have the representation property, which you should set instead (you'll have to decide whether the brokerId or the host is the unique identifier).